pure3.md

```diff
@@ -0,0 +1,225 @@
+default functioning of Uptane, and serve the metadata to the automobile. The
+automobile fetches the image itself from the Image repository along with its
+corresponding in-toto metadata.
+
+### Full Verification ECUs
+
+The Primary ECU receives images and metadata from the repositories, and sends
+them on to the other ECUs that it updates. Each Full Verification ECU is
+extended to perform independent in-toto verification that will be secure even
+if the Primary ECU is compromised. Some Full Verification ECUs, for example the
+Primary, are also expected to perform in-toto verification on behalf of Partial
+Verification ECUs that are resource constrained.
+
+For every target image the client needs to install, the corresponding in-toto
+layout, links, and keys are identified via the custom field of the existing
+Uptane metadata. These files are available as targets on the Image repository.
+Therefore, for each installable target, the updater will need to download and
+verify the Uptane metadata that signs for the in-toto metadata, either the
+Targets role or a delegated role. If verification passes, the actual in-toto
+metadata files—layout, links, and public keys—must be downloaded and their
+hashes must be compared with the Uptane metadata. With these two checks,
+Uptane’s secure delivery properties are extended to the in-toto files. These
+files can then be used with confidence by the client to perform in-toto
+verification.
+
+### Resource Constrained ECUs
+
+The addition of in-toto metadata increases the amount of overhead, which can be
+problematic for resource constrained ECUs. Uptane already accounts for resource
+constrained secondary ECUs by providing a separate Partial Verification
+workflow. Scudo assumes that Uptane Partial Verification ECUs are incapable of
+performing in-toto verification. However, there may be Uptane Full Verification
+ECUs that also lack the resources to perform the added in-toto verification.
+This document specifies that other capable ECUs, perhaps the Primary ECU, must
+perform in-toto verification on behalf of the resource constrained ECUs. In this
+model, while every ECU may not be verifying the software supply chain of the
+images it is expected to install, verification is still happening on the
+individual automobile. Essentially, there is no difference in the update
+workflow of resource constrained ECUs but they are still protected to a degree
+by the supply chain verification happening on the automobile.
+
+It is important to note that some automobiles may have no ECUs capable of
+providing even a basic level of software supply chain security. Therefore, this
+PURE presents a "stop-gap" option. In this configuration, the repository
+verifies the software supply chain of the image and signs an in-toto _summary
+link_ with the keys used to sign either the top level Targets or delegated
+signer role. This way, the key used to sign the summary link metadata is
```

Comment:
Odd choice: why not the same keys as the in-toto root layouts? Also, will there be a layout at all in this case?

Reply:
The verification is performed by the Image repository which is why we pointed to keys that directly map to the repository.